packages/react-devtools-shared/src/backend/utils.js

```diff
@@ -289,3 +290,34 @@ export const isReactNativeEnvironment = (): boolean => {
   // We should probably define the client for DevTools on the backend side and share it with the frontend
   return window.document == null;
 };
+
+export function parseSourceFromComponentStack(
+  componentStack: string,
+): Source | null {
+  const frames = componentStack.split('\n');
+  // eslint-disable-next-line no-for-of-loops/no-for-of-loops
+  for (const frame of frames) {
+    const openingBracketIndex = frame.lastIndexOf('(');
+    if (openingBracketIndex === -1) continue;
+    const closingBracketIndex = frame.lastIndexOf(')');
+    if (
+      closingBracketIndex === -1 ||
+      openingBracketIndex >= closingBracketIndex
+    )
+      continue;
+
+    const url = frame.slice(openingBracketIndex + 1, closingBracketIndex);
+    const match = url.match(/^(.+):(\d+):(\d+)$/);
+    if (match == null) continue;
+
+    const [, sourceURL, line, column] = match;
+
+    return {
+      sourceURL,
+      line: parseInt(line, 10),
+      column: parseInt(column, 10),
+    };
+  }
+
+  return null;
+}
```

Comment:
Should probably remove custom implementation and use `error-stack-parser` for it instead.